src/server/status/index.js

```diff
@@ -18,7 +19,11 @@ export default function (kbnServer, server, config) {
   server.route(wrapAuth({
     method: 'GET',
     path: '/api/status',
-    handler: function (request, reply) {
+    handler: async function (request, reply) {
+      const stats = getStats(
+        server.config().get('kibana.index'),
+        request,
+        server.plugins.elasticsearch.getCluster('admin').callWithRequest);
```

Comment:
I think we want to use `callWithInternalUser` instead of `callWithRequest`. This might prove to be a bit of a headache though, since `getStats` relies on `savedObjectsClient` which expects `callWithRequest`

Because this uses `callWithRequest`, it takes the credentials of the user that calls the API, and requires that they have `read` access to the `.kibana` index (if X-Pack Security is set up).

When I call the API with a user that has no roles, I get this in the logs:
```
server    log   [18:26:09.028] [warning] Detected an unhandled Promise rejection.
Error: action [indices:data/read/search] is unauthorized for user [no_role]
```